api/services/tools/mcp_tools_mange_service.py

```diff
@@ -0,0 +1,225 @@
+import hashlib
+import json
+from datetime import datetime
+
+from sqlalchemy import or_
+from sqlalchemy.exc import IntegrityError
+
+from core.helper import encrypter
+from core.mcp.error import MCPAuthError, MCPConnectionError
+from core.mcp.mcp_client import MCPClient
+from core.tools.entities.api_entities import ToolProviderApiEntity
+from core.tools.entities.common_entities import I18nObject
+from core.tools.entities.tool_entities import ToolProviderType
+from core.tools.mcp_tool.provider import MCPToolProviderController
+from core.tools.utils.configuration import ProviderConfigEncrypter
+from extensions.ext_database import db
+from models.tools import MCPToolProvider
+from services.tools.tools_transform_service import ToolTransformService
+
+UNCHANGED_SERVER_URL_PLACEHOLDER = "[__HIDDEN__]"
+
+
+class MCPToolManageService:
+    """
+    Service class for managing mcp tools.
+    """
+
+    @staticmethod
```

Comment:
I recommend adding an `__init__` method like the following, and turn all static methods into instance methods.

```py
def __init__(self, session: Session):
    self._session = session
```

Then the methods below can use `self._session` instead of `db.session` to query database.

Besides, I recommend moving transaction management out of the service (Unless it is really necessary.) For HTTP requests, the transaction generally should be managed by api handlers.